### DISCUSSION TITLE
Application descriptor - autoshift versionNumber

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#2098] [Q&A]
Is there a way to autoshift versionNumber in app descriptor by 0.001 while publishing app from Animate?

**Comment by ajwfrost** [comment]
The only way I know of to automatically do things is via JSFL:
https://helpx.adobe.com/animate/using/automating-tasks-commands-menu.html

but I don't know whether that would allow you to modify the application descriptor file like this?

Maybe look at 
https://help.adobe.com/archive/en_US/flash/cs5/flash_cs5_extending.pdf
and the `document.publish()` method.. can't see anything about the descriptor file or version code but you may be able to do things using the `FLfile` object..?

**Reply by MalacTheLittle** [reply]
Wow! This hint is quite something! I was totally not aware of JSFL extension. I see that much more automation can be done via this :) Thanks @ajwfrost!

Now, I'm facing an issue here.
I have put together a script for shifting versionNumber, basically it reads descriptor file of active document, shifts version, saves new data to descriptor file and starts document test (document.testMovie()).

The problem is that new version isn't saved to descriptor file each time. I tried to wrap FLfile.write in try-catch and I don't get any errors so can't be 100% sure, but this is probably due to file is opened with other application as I discovered while testing non-JSFL script solutions.
Strange thing is that even if test movie is terminated, it is still occupied by some application for 5-20 seconds (duration is always different, can't figure out the logic). And if test movie isn't terminated but runs for couple of minutes, descriptor file is updated successfully.

Therefore I came up with semi-solution to write shifted version number to separate file, so each time script runs it reads latest version from separate file, shifts it and tries to write it to descriptor file.
Anyway, I'm attaching .jsfl file [here](https://github.com/airsdk/Adobe-Runtime-Support/files/9438626/Test.Movie.version.shift.zip)...


Do you have any clue what is going on behind the scene, what am I doing wrong or how to get descriptor file to be saved each time?
